src/services/projects.js

```diff
@@ -1115,6 +1132,72 @@ module.exports = class ProjectsHelper {
 				}
 			}
 
+			//check for observation as a task
+			if (
+				model == common.TASKS &&
+				entityType.value === common.OBSERVATION &&
+				entityData.type === common.OBSERVATION
+			) {
+				let observationPath = sourceType == '' ? `${common.OBSERVATION}` : `${sourceType}.${common.OBSERVATION}`
+				// Validate the parent solution externalId is present
+				if (!entityData.external_id || entityData.external_id.trim() === '') {
+					validationErrors.push(
+						utils.errorObject(
+							observationPath,
+							common.EXTERNAL_ID,
+							requiredValidation.message || `Required ExternalId${model}`
+						)
+					)
+				}
+				// Validate published externalId
+				if (entityData.external_id && entityMapping[common.OBSERVATION]?.validations) {
+					let consumptionServiceUrl = consumptionConfig.fetchConsumptionServiceUrls(common.OBSERVATION)
+					if (!consumptionServiceUrl) {
+						return responses.failureResponse({
+							message: 'CONSUMPTION_LINK_NOT_FOUND',
+							statusCode: httpStatusCode.bad_request,
+							result: {},
+						})
+					}
+					// Override for Sunbird
+					if (process.env.CONSUMPTION_SERVICE === common.SUNBIRD) {
+						consumptionServiceUrl = process.env.INTERFACE_SERVICE_HOST
+					}
+					const url = utils.buildUrl(consumptionServiceUrl, endpoints.DB_FIND, {}, 'solutions')
+
+					// Body for dbFind
+					const payload = {
+						query: {
+							externalId: entityData.external_id,
+							type: common.OBSERVATION,
+							isReusable: common.TRUE,
+						},
+					}
+
+					const response = await requests.post(url, payload, userDetails.token, true)
+
+					if (!response.success || !response.data) {
+						return responses.failureResponse({
+							message: 'DB_FIND_FAILED',
+							statusCode: httpStatusCode.internal_server_error,
+						})
+					}
+
+					const results = response.data?.result || []
+					// Validate that the solution exits in samiksha service
+					// Only perform this check if the consumption service is not SELF
+					if (process.env.CONSUMPTION_SERVICE !== common.SELF && (results.length === 0 || !results[0]._id)) {
```

Comment:
@praveenKDass if consumption service is self we dont need to make this api call itself